iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/util/BeanMetadataExtraction.java

```diff
@@ -0,0 +1,388 @@
+                entryMetadata.set(HttpHeaders.CONTENT_TYPE, mimeType);
+                entryMetadata.set(ExtraProperties.DECODED_DATA, Boolean.TRUE.toString());
+                entryMetadata.set("bean:className", bean.getClass().getCanonicalName());
+
+                ArrayList<ChildParams> children = new ArrayList<ChildParams>();
+
+                Object[] colObj = null;
+                if (bean instanceof Collection) {
+                    colObj = ((Collection) bean).toArray();
+                }
+                if (bean.getClass().isArray()) {
+                    colObj = (Object[]) bean;
+                }
+                if (bean instanceof Map) {
+                    colObj = ((Map) bean).entrySet().toArray();
+                }
+
+                if (colObj == null) {
+                    String resolvedNameProp = null;
+                    if (bean instanceof Entry) {
+                        resolvedNameProp = "key";
+                    } else {
+                        resolvedNameProp = nameProperties.get(bean.getClass());
+                        if (resolvedNameProp == null) {
+                            resolvedNameProp = nameProperty;
+                        }
+                    }
+                    String resolvedReferencedQuery = referencedQuery.get(bean.getClass());
+                    if (resolvedReferencedQuery != null) {
+                        entryMetadata.add(ExtraProperties.LINKED_ITEMS, parseQuery(resolvedReferencedQuery, bean));
+                    }
+
+                    List<String[]> transformations = getTransformationMapping(bean.getClass());
+                    if (transformations != null) {
+                        for (Iterator iterator = transformations.iterator(); iterator.hasNext();) {
+                            String[] strings = (String[]) iterator.next();
+                            entryMetadata.add(strings[0], parseQuery(strings[1], bean));
+                        }
+                    }
+
+                    for (PropertyDescriptor pd : Introspector.getBeanInfo(bean.getClass()).getPropertyDescriptors()) {
+                        List<String> exclProps = excludeProperties.get(bean.getClass());
+                        if (exclProps != null && exclProps.contains(pd.getName())) {
+                            continue;
+                        }
+                        if (pd.getReadMethod() != null && !"class".equals(pd.getName())) {
+                            Object value = null;
+                            try {
+                                value = pd.getReadMethod().invoke(bean);
+                            } catch (Exception e) {
+                                e.printStackTrace();
+                                continue;
+                            }
+
+                            if (pd.getDisplayName().equals(resolvedNameProp)) {
+                                String name = value.toString().replace("/", "_");
```

Comment:
Is this '/' replacement for '_' needed? We already have items with slashes in their names e.g. emails with dates in their subjects (used as item names)